### PR TITLE
add new catching and throwing overrides for runnable

### PR DESCRIPTION
**Pull request by jokerlee**

### CODE REVIEW

File: src/main/java/com/github/phantomthief/util/MoreFunctions.java [hunk 1]
```diff
@@ -22,10 +22,24 @@
...
+
     public static <R> R throwing(Callable<R> callable) {
         return catching(callable, Throwables::propagate);
     }
 
+    public static <X extends Exception> void throwing(ThrowableRunnable<X> callable) {
+        catching(() -> {
+            callable.run();
+            return null;
+        }, Throwables::propagate);
```
Comment (PhantomThief) [inline src/main/java/com/github/phantomthief/util/MoreFunctions.java]: propagate is deprecated?

File: src/main/java/com/github/phantomthief/util/MoreFunctions.java [hunk 2]
```diff
@@ -22,10 +22,24 @@
         return catching(callable, e -> logger.error("", e));
     }
 
+    public static <X extends Exception> void catching(ThrowableRunnable<X> callable) {
```
Comment (PhantomThief) [inline src/main/java/com/github/phantomthief/util/MoreFunctions.java]: ThrowableRunnable<? extends Throwable>
no need generic type X?

File: src/main/java/com/github/phantomthief/util/MoreFunctions.java (hunk 1 above)
Comment (jokerlee) [inline src/main/java/com/github/phantomthief/util/MoreFunctions.java]: pom dependency is still guava 19.0

File: src/main/java/com/github/phantomthief/util/MoreFunctions.java (hunk 2 above)
Comment (jokerlee) [inline src/main/java/com/github/phantomthief/util/MoreFunctions.java]: just reuse catching(Callable), callable only throws exception, so up bound is Exception

**Pull request merged**